styleguide/git/branching-model.md

```diff
@@ -0,0 +1,30 @@
+# Git Branching Model
+
+Joinbox uses a more specific version of the git-flow branching model
+
+![Joinbox Git Flow Model](git-flow.png "Joinbox Git Flow")
+
+
+## General
+
+- the master branch can be released at any time, it is always 100% stable
+- the develop branch contains only code that works and is stable but not yet tested by the client
+- features and bug fixes are developed in feature branches until they are stable. the base branch for creating feature branches is the release branch where the issue is in or the master branch for hot fixes.
+- feature branches may only be merged into the release branch if they are tested and work correctly.
+- feature branches will only be merged in to the according branch after at least one reviewer approved the pull request.
+- for each iteration (release cycle, milestone) a new release branch is created from develop branch when the development starts.
+- the release branch is published to a testing server for acceptance tests by the client. the client should receive the change log of the release branch
+- fixes for the release branch are merged into the release branch
+- the release branch may be merged into develop or another release branch if it is required
+- hot fixes are only created if they must be applied to the master branch for immediate release. they are based on the master branch
+- after a hot fix or release was released the master branch is merged into the develop branch as well as open release branches
+- releases must be planned in advance, there should not be more than one release branch at any time
+- after the release branch was merged into master and tagged with the correct version the release branch may be deleted.
```

Comment:
dont mention this here `and tagged with the correct version`. this is a separate item: `If the project has version numbers, the correct version number must be set after the release branch was merged into the master branch.`

Reply:
Do you think we should make it optional to use version numbers for a project?